app/controllers/api/v1/publications_controller.rb

```diff
@@ -30,6 +30,18 @@ def grants
       render json: API::V1::GrantSerializer.new(publication.grants)
     end
 
+    def update_all
+      if api_token.write_access?
+        if update_ss_oa_url.positive?
+          render json: { message: I18n.t('api.publications.patch.success'), code: 200 }, status: 200
+        else
+          render json: { message: I18n.t('api.publications.patch.no_content'), code: 404 }, status: 404
```

Comment:
You would get this response if one of the updates failed. And, in that case, any other updates would be rolled back. I'm wondering if the message should say something to that effect. I don't know who the intended user is for this endpoint. Would they want to know that the updates didn't work? Maybe it doesn't matter... but I'm just bringing up the point here. If the updates are rolled back, do we care?

Reply:
I think it would also be a little unclear to me as an API user what would happen if I provided both an Activity Insight ID and a DOI in either the case where those parameters were for the same publication or the case where they were for different publications. It seems to me like the intention is to update one representation of a publication at a time, and so it would make more sense to me if a user could provide _either_ a DOI _or_ an Activity Insight ID, but not both. Right now it looks like our code will accept both parameters at once for two different publications, and it will update both publications with the same URL. I kind of think we should change that so that it's only one of those parameters per request. And then I think all that we could really do to make this behavior more clear to users is to be explicit about it in the documentation and then respond with an error if both params are sent at once.